jolt-core/src/poly/commitment/dory.rs

```diff
@@ -0,0 +1,568 @@
+    fn prove(
+        setup: &Self::Setup,
+        poly: &MultilinearPolynomial<Self::Field>,
+        opening_point: &[Self::Field],
+        transcript: &mut ProofTranscript,
+    ) -> Self::Proof {
+        // extract as JoltField transmutable to arkworks Fr
+        let field_coeffs = extract_field_coefficients(poly);
+
+        // Convert to arkworks Fr
+        let coeffs: Vec<Fr> = field_coeffs
+            .iter()
+            .map(|&coeff| unsafe { std::mem::transmute_copy(&coeff) })
+            .collect();
+
+        // Convert opening point to arkworks Fr
+        let point: Vec<Fr> = opening_point
+            .iter()
+            .map(|&p| unsafe { std::mem::transmute_copy(&p) })
+            .collect();
+
+        let num_vars = poly.get_num_vars();
+        let sigma = (num_vars + 1) / 2;
+
+        let dory_transcript_prover = JoltToDoryTranscript::new(transcript.clone());
+
+        let (evaluation, proof_builder) =
+            evaluate::<ArkBn254Pairing, _, OptimizedMsmG1, OptimizedMsmG2>(
+                &coeffs,
+                &point,
+                sigma,
+                &setup.prover_setup,
+                dory_transcript_prover,
+            );
+
+        // Extract from the builder into a serializable form
+        let dory_proof = proof_builder.build();
+
+        DoryProof {
+            evaluation,
+            opening_point: point,
+            sigma,
+            dory_proof_data: dory_proof,
+            _phantom: PhantomData,
+        }
+    }
+
+    fn verify(
+        proof: &Self::Proof,
+        setup: &Self::Setup,
+        transcript: &mut ProofTranscript,
+        opening_point: &[Self::Field],
+        _opening: &Self::Field, // we use the opening directly from Dory's Proof object.
+        commitment: &Self::Commitment,
+    ) -> Result<(), ProofVerifyError> {
+        
+        // Convert opening point to arkworks Fr
+        let point: Vec<Fr> = opening_point
+            .iter()
+            .map(|&p| unsafe { std::mem::transmute_copy(&p) })
+            .collect();
+
+        // Create JoltToDoryTranscript wrapper around the provided transcript
+        let dory_transcript = JoltToDoryTranscript::new(transcript.clone());
+
+        // Create a proof builder from the DoryProof
+        let verifier_builder =
+            DoryProofBuilder::from_proof(proof.dory_proof_data.clone(), dory_transcript.clone());
+
+        // Use Dory's verify function with the proof builder
+        let verify_result =
+            verify::<ArkBn254Pairing, _, OptimizedMsmG1, OptimizedMsmG2, DummyMsm<Fq12>>(
+                commitment.commitment,
+                proof.evaluation,
+                &point,
+                verifier_builder,
+                proof.sigma,
+                &setup.verifier_setup,
+                dory_transcript,
+            );
+
+        match verify_result {
+            Ok(()) => Ok(()),
+            Err(_) => Err(ProofVerifyError::InternalError),
```

Comment:
Maybe add more context for this error, or a custom one

Reply:
This point still holds, would help debugging to know which of three options of DoryError it is